sklearn/linear_model/coordinate_descent.py

```diff
@@ -69,6 +69,11 @@ def _alpha_grid(X, y, Xy=None, l1_ratio=1.0, fit_intercept=True,
     copy_X : boolean, optional, default True
         If ``True``, X will be copied; else, it may be overwritten.
     """
+    if np.unique(y).size == 1:
```

Comment:
alpha_max is zero, only when `fit_intercept` is set to True (and when all n_targets) are the same.  This block of code should be inside a `if fit_intercept` condition, IMO